### ISSUE TITLE
Edit controls: Selection not reported if selection changes but caret doesn't move

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#1898]
**Reported by ianr on 2011-11-10 17:10**
I have noticed that if I have 2 lines of text in notepad I get strange behavior when performing the following commands.

1 Press control + end to go to the end of the text
2 press control + shift + home to select all the text
3 press control + home
At this point NVDA read the current line, which is fine for me.  The text also becomes unselected though NVDA does not announce this.

4 Press right arrow or I think any text navigation key.
At this point NVDA reads the character you moved to and then announces that the lines have become unselected.
This is incorrect because they actually became unselected after step 3.

The braille output also portrays the same behavior.

**Comment by jcsteh** [comment]
P2 because this misleads users and could result, for example, in text being removed accidentally.

- Are there any other ways we might detect selection changes for Win32 Edit controls?
- Failing that, we might have to use EditableTextWithoutAutoSelectDetection for Edit controls (but not RichEdit).
    - We didn't want to use this originally because that would mean things like mouse selection, typing over the selection, etc. wouldn't be reported. Is there some way we can get the best of both worlds without a conflict?

@MichaelDCurran, AUD$0.01 for your thoughts?

**Comment by Adriani90** [comment]
@michaelDCurran, any updates on this issue? It is still reproducible i.e. after selecting the text and pressing ctrl+up arrow followed by right arrow.

**Comment by CyrilleB79** [comment]
Cannot reproduce on Win 10 1709's notepad.
Is it a Win7 only bug?

**Comment by ehollig** [comment]
Echoing what @CyrilleB79 said. I can not reproduce with NVDA 2019.2 on Windows 10,1709. Is this specific to Windows 7?

**Comment by Adriani90** [comment]
@ehollig, @CyrilleB79 it is not reproducible with step 3 in the description. But if you press ctrl+up arrow after selecting text, it is still reproducible. It seems specific to ctrl+up and down arrow.
cc: @lukaszgo1, @XLTechie

**Comment by XLTechie** [comment]
@Adriani90 I am unable to replicate this, either with ianr's method, or with your method, under Windows 10 (1903), NVDA 2019.2. Unfortunately, I no longer have Windows 7 machines available. I can test with older NVDA if desired for some reason.

Specifically, for your method, I did the following:

1. In notepad, I entered the following text:
```
line 1
line 2
line 3
line 4
line 5
```
2. I moved to "line 4".
3. I pressed shift+up arrow twice, to select lines 3 and 4.
4. I then pressed control+up arrow several times.
    * Each time, I heard "line 2" spoken.
5. I then pressed control+c.
6. I moved to the bottom of the file, made a few blank lines, and then pressed control+v.
7. What was pasted, was lines 3 and 4, as I expected.
8. When I tried this again, and instead of copying, pressed right or left arrow, I heard "Line 3, line 4, unselected" as I expected.

My conclusion is that control+up/down arrow, has no effect in Notepad or NVDA, and does not alter selection, which is the expected behavior.

I am happy to test in any other way you might think of.
But at the moment my opinion is that this issue should be closed.

**Comment by XLTechie** [comment]
I commented a few minutes ago, but deleted it. Revised version here.

@Adriani90 I am unable to replicate this, in Notepad, as described in Ianr's method (the original issue).
However, I can replicate both Ianr's problem, and what you describe, under Word 2019 (no UIA).
Win 10 (1903), NVDA 2019.2.1 and 2019.3 latest alpha.

Specifically, for your method, I did the following:

1. In Notepad, and Word, I entered the following text:
```
line 1
line 2
line 3
line 4
line 5
```
2. I moved to "line 4", and pressed end.
3. I pressed shift+up arrow twice, to select lines 3 and 4.
4. I then pressed control+up arrow several times.
    * In Notepad: each time, I heard "line 2" spoken, and the selection didn't change.
    * In Word: I heard "line 2", then "line 1". The selection was undone, but nothing was spoken about the selection or unselection.
5. I then pressed control+c.
    * In Notepad: lines 3 and 4 were copied.
    * In Word: I heard an error sound, and nothing was copied.
6. In Notepad, I also tried pressing right arrow or left arrow, instead of copying in step 5.
    * I heard "line 3, line 4, unselected", which is expected.

My conclusion is that control+up/down arrow, has no effect in Notepad or NVDA, and does not alter Notepad selection, which is the expected behavior.
However in Word, it definitely does alter the selection (unselecting everything), and just as definitely NVDA doesn't say anything about it.

Bug needing work, with seriousness as noted in https://github.com/nvaccess/nvda/issues/1898#issuecomment-303896016

**Comment by XLTechie** [comment]
I vote for renaming to be Word-specific, and leaving open. The original problem 
is solved if taken literally, but the systemic problem has not been, and this 
issue has all the context. No reason to write it all up again somewhere else.

**Comment by Adriani90** [comment]
@CyrilleB79 I see the point. I do not have enough sight to see if the selection was removed in notepad. I thought it was but now I understand. To avoid confusions, could you please open a new issue specific to MS word including the visual aspects of this? Honnestly I lean towars closing this one to avoid off topic comments.

**Comment by CyrilleB79** [comment]
All what is described in initial description of this issue has cannot be reproduced. I recommand closing this ticket. @Adriani90 can you do it?

Regarding the off-topic discussion about MS Word, I have opened a new issue to have it tracked: #10518. I have simplified the steps to reproduce since the issue is only that unselecting text is not reported in MS Word on contrary to Notepad or Wordpad. The visual feedback of selection was only here as a confirmation, but there is no need to have some vision left to analyse this issue.

**Comment by Adriani90** [comment]
Closing as works for me.
